maindata.py

```diff
@@ -4,32 +4,44 @@
 from csvoutput import CsvOutput
 from wikipediafunction import WikipediaFunction
 from youtubefunction import YoutubeFunction
+from steamfunction import SteamFunction
+#import time
+from multiprocessing.pool import ThreadPool
+
+pool = ThreadPool(processes=1)
 
 redditFunction = RedditFunction()
 imdbFunction = ImdbFunction()
 csvOutput = CsvOutput()
 wikipediaFunction = WikipediaFunction()
 youtubeFunction = YoutubeFunction()
+steamFunction = SteamFunction()
 
 #input string
 q = 'gravity'
 
-#adapt string for reddit or imdb query
-queryr = q.replace(" ","+")
-queryi = q.replace(" ","-")
-queryw = q.replace(" ","+")
-queryy = q.replace(" ","+")
+#tic = time.perf_counter()
+
+#Run reddit in seperate thread to reduce execution time
+rThread = pool.apply_async(redditFunction.getPushshiftData, (100, 1526428800,1589587200, q))
+
+i = imdbFunction.getIMDB(q)
+w = wikipediaFunction.getWiki(10, q)
+y = youtubeFunction.getYouTube(q)
+s = steamFunction.getSteam(q)
+
+#get thread running for reddit's output
+r = rThread.get()

```

Comment:
Not sure it's easier, but Python 3.6 added concurrent > futures which makes it pretty easy to thread things - https://docs.python.org/3/library/concurrent.futures.html - just an FYI

Reply:
I implemented this (see below code snippet) and the output appears to be much slower than using the multiprocessing.pool _(on average it takes roughly 5 seconds more, which is still an improvement from no multi threading support)_ 

I was pretty surprised by this, so did a quick google search and found [this on stack overflow](https://stackoverflow.com/questions/18671528/processpoolexecutor-from-concurrent-futures-way-slower-than-multiprocessing-pool) 

``` python
# -*- coding: utf-8 -*-
from redditfunction import RedditFunction
from imdbfunction import ImdbFunction
from csvoutput import CsvOutput
from wikipediafunction import WikipediaFunction
from youtubefunction import YoutubeFunction
from steamfunction import SteamFunction
import time
from concurrent.futures import ThreadPoolExecutor

redditFunction = RedditFunction()
imdbFunction = ImdbFunction()
csvOutput = CsvOutput()
wikipediaFunction = WikipediaFunction()
youtubeFunction = YoutubeFunction()
steamFunction = SteamFunction()

#input string
q = 'gravity'
size = 10
tic = time.perf_counter()

#Run reddit in seperate thread to reduce execution time
with ThreadPoolExecutor(max_workers=1) as executor:
    rThread = executor.submit(redditFunction.getPushshiftData, 100, 1526428800,1589587200, q, size)
    
i = imdbFunction.getIMDB(q, size)
w = wikipediaFunction.getWiki(q, size)
y = youtubeFunction.getYouTube(q, size)
s = steamFunction.getSteam(q, size)

#get thread running for reddit's output
r = rThread.result()

toc = time.perf_counter()
print(f"did the thing in {toc - tic:0.4f} seconds")

#URL lists
print(r)
print(i)
print(w)
print(y)
print(s)
mylist = r + i + w + y + s

csvOutput.csvwrite(mylist, q)
```